substrate/frame/vesting/src/lib.rs

```diff
@@ -678,6 +686,119 @@ impl<T: Config> Pallet<T> {
 	}
 }
 
+/// Ensure the correctness of the state of this pallet.
+///
+/// The following expectations must always apply.
+///
+/// ## Expectations:
+///
+/// `handle_before_schedule_starts`:
+/// * the locked amount of a vesting schedule must be equal to the product of the duration
+///   (`schedules_left` - `starting_block`) and the per block amount when the locked amount is
+///   divisible by the per block amount.
+/// * However, If the locked amount is not divisible by the per block amount, the final vesting
+///   block (`schedules_left` - 1), the unvested amount should be equal to the remainder.
+///
+/// `handle_during_schedule`:
+/// * the amount `still_vesting` must be equal to the product of the remaining blocks to vest
+///   (`schedules_left` - `current_block`) and per block amount when the locked amount is divisible
+///   by the per block amount.
+/// * However, If the locked amount is not divisible by the per block amount, then at the final
+///   vesting block of the current schedule (`schedules_left` - 1), the unvested amount should be
+///   equal to the remainder.
+#[cfg(any(feature = "try-runtime", test))]
+impl<T: Config> Pallet<T> {
+	pub fn do_try_state() -> Result<(), TryRuntimeError> {
+		for (_, d) in Vesting::<T>::iter() {
```

Comment:
Please do the same thing as in the `tips` pallet with the `iter_keys` to ensure that there is no undecodable data.

Reply:
yes, thanks for the review